include/uapi/sound/sof/header.h

```diff
@@ -26,6 +26,30 @@ struct sof_abi_hdr {
 	__u32 data[];		/**< Component data - opaque to core */
 }  __packed;
 
+/*
+ * Identifies data type, size and ABI for IPC4.
+ * Used by any bespoke component data structures or binary blobs.
+ */
+struct sof_ipc4_abi_hdr {
+	__u32 magic;		/**< 'S', 'O', 'F', '4' */
+	__u32 rsvd1;
+	__u32 size;		/**< size in bytes of data excl. this struct */
+	__u32 abi;		/**< SOF ABI version */
+	__u32 ipc_blob_type;	/** IPC4 type. 0: init, 2: config, 4: large_config */
+	/*
+	 * ipc_extension is used in ipc4 message extension field. Please check
+	 * struct sof_ipc4_msg for the ipc4 message structure details.
+	 * ipc_extension in this structure is used by module config/large_config
+	 * ipc message. Module config message uses extension to send the configure
+	 * data while module large_config message uses extension to describe
+	 * the message payload size, param_id, etc. If the extension is not
+	 * used, it should be set to 0.
+	 */
+	__u32 ipc_extension;
+	__u32 rsvd2[2];	/**< reserved for future use */
+	__u32 data[];		/**< Component data - opaque to core */
```

Comment:
is there not a size here? How would the consumer of this abi_hdr know how many bytes to use?

Reply:
Thie field ```__u32 size; /**< size in bytes of data excl. this struct */``` tells the data size.

Reply:
then call it data_size or something and move it just above data[].... and put your reserved things in the middle.

Reply:
@plbossart As @ujfalusi @ranj063 suggested, I reused the sof_abi_hdr for IPC4 support now and no sof_ipc4_abi_hdr anymore.

Reply:
@libinyang, so what is the catch with the `sof_abi_hdr.type` VS `sof_abi_hdr.ipc4.ipc_blob_type` ?
Are we ever going to use `sof_abi_hdr.type` for IPC4 and if not then why not re-use it for the same purpose as the `ipc_blob_type`?
If both will be used than I would rename the `ipc_blob_type` to something else, blob_message_type, blob_ipc_type (but I don't like the ipc in there), it is an artificial number for the kernel to know what kind of message it should use to set/get the blob to/from the firmware, right?

Reply:
and you should update the documentation for the SOF4 magic, probably moving them to kernel style doc as I did.

Reply:
@libinyang, my question regarding to `sof_abi_hdr.type` VS `sof_abi_hdr.ipc4.ipc_blob_type` still valid.